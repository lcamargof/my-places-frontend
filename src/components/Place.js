import React from 'react';
import PropTypes from 'prop-types';
import EditIcon from '@material-ui/icons/Edit';
import classNames from 'classnames';
import moment from 'moment';
import RemoveIcon from '@material-ui/icons/Delete';
import { IconButton, withStyles } from 'material-ui';

const styles = () => ({
  root: {
    width: '180px'
  },
  title: {
    textAlign: 'center',
    marginTop: 0,
  },
  open: {
    color: 'green',
  },
  close: {
    color: 'red',
  },
  button: {
    curosr: 'pointer',
  },
  btnContainer: {
    textAlign: 'center',
    marginBottom: '-10px',
  },
  description: {
    marginBottom: 0,
    borderBottom: '1px solid #ccc',
    paddingBottom: '10px',
    fontSize: '14px',
    wordWrap: 'break-word',
  },
  status: {
    textAlign: 'center',
  },
});

export const Place = ({ classes, place, onDelete, onEdit }) => {

  const open = moment().isBetween(moment(place.open, 'HH:mm'), moment(place.close, 'HH:mm'), 'hours', '[]');

  return (
    <div className={classes.root}>
      <h2 className={classNames(classes.title, open ? classes.open : classes.close)}>
        {place.name}
      </h2>
      <p className={classes.description}>{ place.description }</p>
      <div className={classes.btnContainer}>
        <IconButton id="place-edit" onClick={onEdit} className={classes.button} aria-label="Edit">
          <EditIcon />
        </IconButton>
        <IconButton id="place-delete" onClick={onDelete} className={classes.button} aria-label="Delete">
          <RemoveIcon />
        </IconButton>
      </div>
    </div>
  )
};

Place.propTypes = {
  place: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  onDelete: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
};

export default withStyles(styles)(Place);