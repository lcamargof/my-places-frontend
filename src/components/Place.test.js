import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Place } from './Place';

describe('Component: Place', () => {
  let component = null;
  var classes, place, onEdit, onDelete;

  beforeEach(() => {
    classes = {};
    place = { name: 'test', open: '00:00', close: '23:59', description: '' };
    onDelete = jest.fn();
    onEdit = jest.fn();
  });

  const buildComponent = () => {
    const props = {
      classes, place, onEdit, onDelete
    };

    return shallow(<Place {...props} />);
  };

  describe('Snapshots', () => {
    it('Should render correctly', () => {
      component = buildComponent();
      expect(shallowToJson(component)).toMatchSnapshot();
    });
  });

  describe('Actions', () => {
    it('Should trigger edit when clicked the edit button', () => {
      component = buildComponent();
      component.find('#place-edit').simulate('click');
      expect(onEdit).toHaveBeenCalledTimes(1);
    });

    it('Should trigger delete when clicked the delete button', () => {
      component = buildComponent();
      component.find('#place-delete').simulate('click');
      expect(onDelete).toHaveBeenCalledTimes(1);
    });
  });
});