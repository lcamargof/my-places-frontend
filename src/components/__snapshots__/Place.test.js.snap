// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Place Snapshots Should render correctly 1`] = `
<div>
  <h2
    className=""
  >
    test
  </h2>
  <p />
  <div>
    <WithStyles(IconButton)
      aria-label="Edit"
      id="place-edit"
      onClick={[Function]}
    >
      <pure(Edit) />
    </WithStyles(IconButton)>
    <WithStyles(IconButton)
      aria-label="Delete"
      id="place-delete"
      onClick={[Function]}
    >
      <pure(Delete) />
    </WithStyles(IconButton)>
  </div>
</div>
`;
